utils/R/convert.input.R

```diff
@@ -6,65 +6,133 @@
 ##' @export
 ##' @author Betsy Cowdery, Michael Dietze, Ankur Desai
 convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, start_date, end_date, 
-                          pkg, fcn, username, con=con, host, browndog, write=TRUE,  
-                          format.vars=format.vars, ...) {
-  l <- list(...); #print(l)
+                          pkg, fcn, con=con, host, browndog, write=TRUE,  
+                          format.vars, overwrite=FALSE, ...) {
+  input.args <- list(...)
+
+  logger.debug(paste(
+    "Convert.Inputs", fcn, input.id, host$name, outfolder, formatname, mimetype, 
+    site.id, start_date, end_date))
 
-  logger.info(paste("Convert.Inputs",fcn,input.id,host$name,outfolder,formatname,mimetype,site.id,start_date,end_date))
   n <- nchar(outfolder)
-  if(substr(outfolder,n,n) != "/"){outfolder = paste0(outfolder,"/")}
-  
-  outname = tail(unlist(strsplit(outfolder,'/')),n=1)
+  if(substr(outfolder,n,n) != "/") {
+    outfolder = paste0(outfolder,"/")
+  }
   
-  print(start_date)
-  startdate <- as.POSIXlt(start_date, tz = "GMT")
-  enddate   <- as.POSIXlt(end_date, tz = "GMT")
+  outname = tail(unlist(strsplit(outfolder,'/')), n=1)
 
-  # Consider adding a force option to skip the check and continue with conversion
-  print("start CHECK")
-  check = dbfile.input.check(site.id, startdate, enddate, mimetype, formatname, parentid=input.id, con=con, host$name)
+  print(paste(
+    "start CHECK Convert.Inputs", fcn, input.id, host$name, outfolder, formatname, mimetype, 
+    site.id, start_date, end_date))
+  existing.dbfile <- dbfile.input.check(
+    siteid=site.id, mimetype=mimetype, formatname=formatname, parentid=input.id, 
+    con=con, hostname=host$name, ignore.dates=TRUE)
+  print(existing.dbfile, digits=10)
   print("end CHECK")
-  print(check)
-  if(length(check)>0){
-    return(list(input.id=check$container_id, dbfile.id=check$id))
+
+  if(nrow(existing.dbfile) > 0) {
+    existing.input <- db.query(paste0("SELECT * FROM inputs WHERE id=", existing.dbfile[['container_id']]), con)    
+    # Convert dates to Date objects and strip all time zones (DB values are timezone-free)
+    start_date <- force_tz(as_date(start_date), 'GMT')
+    end_date <- force_tz(as_date(end_date), 'GMT')
+    existing.input$start_date <- force_tz(as_date(existing.input$start_date), 'GMT')
+    existing.input$end_date <- force_tz(as_date(existing.input$end_date), 'GMT')
+
+    if(overwrite) {
+      # collect files to flag for deletion
+      files.to.delete <- remote.execute.R(paste0(
+        "list.files('", existing.dbfile[['file_path']], "', full.names=TRUE)"), 
+        host, user=NA, verbose=TRUE, R="R")
+
+      file.deletion.commands <- .get.file.deletion.commands(files.to.delete)
+      
+      remote.execute.R(file.deletion.commands$move.to.tmp, host, user=NA, verbose=TRUE, R="R")
+      
+      # Schedule files to be replaced or deleted on exiting the function
+      succesful <- FALSE
```

Comment:
Shit, thanks. This explains a bug I was hitting for awhile and, apparently, fixed some incorrect way.